cmd/copyEnumeratorInit.go

```diff
@@ -102,35 +97,110 @@ func (cca *cookedCopyCmdArgs) initEnumerator(jobPartOrder common.CopyJobPartOrde
+				seenFailedContainers[dstContainerName] = true
+			}
+		} else if cca.fromTo.From().IsRemote() { // if the destination has implicit container names
+			if acctTraverser, ok := traverser.(accountTraverser); ok && dstLevel == ELocationLevel.Service() {
+				containers, err := acctTraverser.listContainers()
+
+				if err != nil {
+					return nil, fmt.Errorf("failed to list containers: %s", err)
+				}
+
+				containerResolver = NewS3BucketNameToAzureResourcesResolver(containers)
+
+				for _, v := range containers {
+					bucketName, err := containerResolver.ResolveName(v)
+
+					if err != nil {
+						// Silently ignore the failure; it'll get logged later.
+						continue
+					}
+
+					err = cca.createDstContainer(bucketName, dst, ctx, existingContainers)
+
+					// if JobsAdmin is nil, we're probably in testing mode.
+					// As a result, container creation failures are expected as we don't give the SAS tokens adequate permissions.
+					// check against seenFailedContainers so we don't spam the job log with initialization failed errors
+					if _, ok := seenFailedContainers[bucketName]; err != nil && ste.JobsAdmin != nil && !ok {
+						fmt.Println(err)
+						logDstContainerCreateFailureOnce.Do(func() {
+							glcm.Info("Failed to create one or more destination container(s). Your transfers may still succeed if the container already exists.")
+						})
+						ste.JobsAdmin.LogToJobLog(fmt.Sprintf("failed to initialize destination container %s; the transfer will continue (but be wary it may fail): %s", bucketName, err))
+						seenFailedContainers[bucketName] = true
+					}
+				}
+			} else {
+				cName, err := GetContainerName(src, cca.fromTo.From())
+
+				if err != nil || cName == "" {
+					// this will probably never be reached
+					return nil, fmt.Errorf("failed to get container name from source (is it formatted correctly?)")
+				}
+
+				resName, err := containerResolver.ResolveName(cName)
+
+				if err == nil {
+					err = cca.createDstContainer(resName, dst, ctx, existingContainers)
+
+					if _, ok := seenFailedContainers[dstContainerName]; err != nil && ste.JobsAdmin != nil && !ok {
+						logDstContainerCreateFailureOnce.Do(func() {
+							glcm.Info("Failed to create one or more destination container(s). Your transfers may still succeed if the container already exists.")
+						})
+						ste.JobsAdmin.LogToJobLog(fmt.Sprintf("failed to initialize destination container %s; the transfer will continue (but be wary it may fail): %s", dstContainerName, err))
+						seenFailedContainers[dstContainerName] = true
+					}
+				}
+			}
+		}
+	}
+
+	filters := cca.initModularFilters()
+	processor := func(object storedObject) error {
+		// Start by resolving the name and creating the container
+		if object.containerName != "" {
+			// set up the destination container name.
+			cName := dstContainerName
```

Comment:
If a destination container name is explicitly specified, do not resolve it, do not add it to the path creation.

If we _did_  use it here, then we'd basically get
https://account.blob.core.windows.net/container/container/dstfile
instead of
https://account.blob.core.windows.net/container/dstfile